butterfly-slack/src/main/java/com/paypal/butterfly/slack/SlackNotifierConfig.java

```diff
@@ -0,0 +1,140 @@
+                for (String id : butterflyUserIDs) {
+                    //TODO: Change hard coded email into butterflyUserIDs
+                    String email = id + ""; //id + "@paypal.com";
+                    String slackUserID = getSlackUserID(target, token, email);
+                    String slackChannelID = getSlackChannelID(target, token, slackUserID);
+                    postSlackMessage(target, token, jsonMessage, slackChannelID);
+                }
+            }
+
+            /**
+             * GET slack user IDs from PayPal email using the users.lookupByEmail web api call
+             */
+            private String getSlackUserID(WebTarget target, String token, String email){
+                    JsonObject slackResponse = new JsonParser().parse(target
+                            .path("users.lookupByEmail")
+                            .queryParam("email", email)
+                            .request(MediaType.APPLICATION_JSON)
+                            .header("Authorization", "Bearer " + token)
+                            .get(String.class))
+                            .getAsJsonObject();
+                    String slackUserID = "";
+                    try {
+                        slackUserID = slackResponse.getAsJsonObject("user").get("id").getAsString();
+                    } catch (Exception e) {
+                        logger.warn("An exception happened when retrieving the slack user id " + slackResponse.get("error").getAsString(), e);
+                    }
+                    return slackUserID;
+            }
+
+            /**
+             * POST to open direct message channel using the im.open web api call
+             */
+            private String getSlackChannelID(WebTarget target, String token, String slackUserID){
+                JsonObject slackResponse = new JsonParser().parse(target
+                        .path("im.open")
+                        .queryParam("user", slackUserID)
+                        .request(MediaType.APPLICATION_JSON)
+                        .header("Authorization", "Bearer " + token)
+                        .get(String.class))
+                        .getAsJsonObject();
+                String slackChannelID = "";
+                try {
+                    slackChannelID = slackResponse.getAsJsonObject("channel").get("id").getAsString();
+                } catch (Exception e) {
+                    logger.warn("An exception happened when retrieving the slack channel id " + slackResponse.get("error").getAsString(), e);
+                }
+                return slackChannelID;
+            }
+
+            /**
+             * POST slack message to all slackIDs using the chat.postMessage web api call
+             */
+            private void postSlackMessage(WebTarget target, String token, String jsonFileString, String slackChannelID){
+                String jsonFileStringFormat = String.format(jsonFileString, slackChannelID);
```

Comment:
Add a try catch here as well

Reply:
no exception is ever thrown here, the client call will always return a json object with {"ok": "true", ...} or {"ok" : "false", ...}